core/src/main/java/org/apache/iceberg/BaseTable.java

```diff
@@ -136,37 +136,37 @@ public RewriteManifests rewriteManifests() {
 
   @Override
   public OverwriteFiles newOverwrite() {
-    return new BaseOverwriteFiles(ops);
+    return new BaseOverwriteFiles(name, ops);
   }
 
   @Override
   public ReplacePartitions newReplacePartitions() {
-    return new BaseReplacePartitions(ops);
+    return new BaseReplacePartitions(name, ops);
   }
 
   @Override
   public DeleteFiles newDelete() {
-    return new StreamingDelete(ops);
+    return new StreamingDelete(name, ops);
   }
 
   @Override
-  public ExpireSnapshots expireSnapshots() {
+  public RemoveSnapshots expireSnapshots() {
```

Comment:
Why did this change? Can we not return the interface?

Reply:
Fixed. My bad.